lib/services/ngsi/entities-NGSI-LD.js

```diff
@@ -0,0 +1,541 @@
+            obj.value = valueOfOrNull(Number.parseInt(attr.value));
+            break;
+        case 'number':
+            if (NGSIUtils.isFloat(attr.value)) {
+                obj.value = valueOfOrNull(Number.parseFloat(attr.value));
+            } else {
+                obj.value = valueOfOrNull(Number.parseInt(attr.value));
+            }
+            break;
+
+        // Temporal Properties
+        case 'datetime':
+            obj.value = {
+                '@type': 'DateTime',
+                '@value': moment.tz(attr.value, 'Etc/UTC').toISOString()
+            };
+            break;
+        case 'date':
+            obj.value = {
+                '@type': 'Date',
+                '@value': moment.tz(attr.value, 'Etc/UTC').format(moment.HTML5_FMT.DATE)
+            };
+            break;
+        case 'time':
+            obj.value = {
+                '@type': 'Time',
+                '@value': moment.tz(attr.value, 'Etc/UTC').format(moment.HTML5_FMT.TIME_SECONDS)
+            };
+            break;
+
+        // GeoProperties
+        case 'geoproperty':
+        case 'point':
+        case 'geo:point':
+            obj.type = 'GeoProperty';
+            obj.value = { type: 'Point', coordinates: getLngLats(attr.value) };
+            break;
+        case 'linestring':
+        case 'geo:linestring':
+            obj.type = 'GeoProperty';
+            obj.value = { type: 'LineString', coordinates: getLngLats(attr.value) };
+            break;
+        case 'polygon':
+        case 'geo:polygon':
+            obj.type = 'GeoProperty';
+            obj.value = { type: 'Polygon', coordinates: getLngLats(attr.value) };
+            break;
+        case 'multipoint':
+        case 'geo:multipoint':
+            obj.type = 'GeoProperty';
+            obj.value = { type: 'MultiPoint', coordinates: getLngLats(attr.value) };
+            break;
+        case 'multilinestring':
+        case 'geo:multilinestring':
+            obj.type = 'GeoProperty';
+            obj.value = { type: 'MultiLineString', coordinates: attr.value };
+            break;
+        case 'multipolygon':
+        case 'geo:multipolygon':
+            obj.type = 'GeoProperty';
+            obj.value = { type: 'MultiPolygon', coordinates: attr.value };
+            break;
+
+        // Relationships
+        case 'relationship':
+            obj.type = 'Relationship';
+            obj.object = attr.value;
+            delete obj.value;
+            break;
+
+        default:
+            obj.value = { '@type': attr.type, '@value': attr.value };
+    }
+
+    if (attr.metadata) {
+        Object.keys(attr.metadata).forEach(function(key) {
+            switch (key) {
+                case constants.TIMESTAMP_ATTRIBUTE:
+                    var timestamp = attr.metadata[key].value;
+                    if (timestamp === constants.ATTRIBUTE_DEFAULT || !moment(timestamp).isValid()) {
+                        obj.observedAt = constants.DATETIME_DEFAULT;
```

Comment:
be careful, constants.ATTRIBUTE_DEFAULT/DATETIME_DEFAULT are not exported in constants.js, this may lead to an undefined observedAt

Reply:
That is a good catch.

@fgalan - you should update `constants.js` to export as shown.

```diff
  MONGO_ALARM: 'MONGO-ALARM',
    ORION_ALARM: 'ORION-ALARM',
    IOTAM_ALARM: 'IOTAM-ALARM',

+    ATTRIBUTE_DEFAULT,
+   DATETIME_DEFAULT,

    getInitialValueForType
};

```

Reply:
Fixed in PR https://github.com/telefonicaid/iotagent-node-lib/pull/882 (which merges into this PR's branch)